src/include/homestore/blkdata_service.hpp

```diff
@@ -52,59 +52,120 @@ class BlkDataService {
     /**
      * @brief : called during recovery to open existing vdev for data service
      *
-     * @param vb
+     * @param vb : vdev info blk containing the details of this blkstore
      */
     void open_vdev(vdev_info_block* vb);
 
     /**
      * @brief : called in non-recovery mode to create a new vdev for data service
      *
-     * @param size
+     * @param size : size of this vdev
      */
     void create_vdev(uint64_t size);
 
+    /**
+     * @brief : asynchronous write without input block ids. Block ids will be allocated by this api and returned;
+     *
+     * @param sgs : the data buffer that needs to be written
+     * @param hints : blk alloc hints
+     * @param out_blkids : the output block ids that were allocated and written to
+     * @param cb : callback that will be triggered after write completes;
+     * @param part_of_batch : is this write part of a batch;
+     */
     void async_write(const sisl::sg_list& sgs, const blk_alloc_hints& hints, std::vector< BlkId >& out_blkids,
                      const io_completion_cb_t& cb, bool part_of_batch = false);
+
+    /**
+     * @brief : asynchronous write with input block ids;
+     *
+     * @param sgs : the data buffer that needs to be written
+     * @param hints : blk alloc hints
+     * @param in_blkids : input block ids that this write should be written to;
+     * @param cb : callback that will be triggered after write completes
+     * @param part_of_batch : is this write part of a batch;
+     */
+    void async_write_ahead(const sisl::sg_list& sgs, const blk_alloc_hints& hints,
+                           const std::vector< BlkId >& in_blkids, const io_completion_cb_t& cb,
+                           bool part_of_batch = false);
+
+    /**
+     * @brief : asynchronous read
+     *
+     * @param bid : block id to read
+     * @param sgs : the read buffer stored
+     * @param size : size to read
+     * @param cb : callback that will be triggered after read completes
+     * @param part_of_batch : is this read part of batch;
+     */
```

Comment:
My suggestion is to rename `async_write_ahead()` to `async_write()` and current `async_write()` to `async_alloc_write()`. Write ahead in general has different meaning right (similar to read_ahead)?

Reply:
I will take your suggestion, let me update the PR.